### PR TITLE
Modifying java servlet

### PR DESCRIPTION
**Pull request by shreyabarua**
Completing Java servlet walkthrough
I'm not sure if this is exactly what the walkthrough was looking for, please let me know if you see anything that looks off.

### CODE REVIEW

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -29,3 +29,45 @@ function showSlides(n) {
...
+  console.log('Handling the response.');
+
+  // response.text() returns a Promise, because the response is a stream of
+  // content and not a simple variable.
+  const textPromise = response.text();
+
+  // When the response is converted to text, pass the result into the
+  // addMsgToDom() function.
+  textPromise.then(addMsgToDom);
+}
```
Comment (astrelni) [inline portfolio/src/main/webapp/script.js]: This is nice and readable. For better or worse I think the more idiomatic way to do this in javascript is more concisely stringing together arrow function expressions like:

```
fetch('/data').then(response => response.text()).then(message => addMsgToDom(message));
```

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -29,3 +29,45 @@ function showSlides(n) {
...
+
+/** Adds message to the DOM. */
+function addMsgToDom(message) {
+  console.log('Adding message to dom: ' + message);
```
Comment (astrelni) [inline portfolio/src/main/webapp/script.js]: Not sure if we should keep all of the logging messages. If you'd like to keep them for now, I guess we can take them out in a more final version of the portfolio.

**Review by astrelni — DISMISSED** [review]
Cool.

Main thing that is unclear to me is that we have the async message function that is getting used, and we also have a non-async function and its helpers that seem like unused code?

File: portfolio/src/main/webapp/script.js (hunk 1 above)
Comment (shreyabarua) [reply portfolio/src/main/webapp/script.js]: that looks good, I changed it to this format

**Comment by shreyabarua** [comment]
I left the async function just so I could practice writing the function both ways. Should I remove the function that I am not using?

**Review by oydang — APPROVED** [review]
You can leave the async function but I'd put a comment above it saying this is unused and untested for now.

**Pull request merged**